### PR TITLE
ensure hhkb runs matrix_init_quantum

### PR DESCRIPTION
**Pull request by joshuarubin**
ensure hhkb runs matrix_init_quantum

## Description

eeprom wasn't being restored on init, this fixes that

## Types of Changes

- [ ] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/hhkb/ansi/matrix.c [hunk 1]
```diff
@@ -69,15 +69,19 @@ void matrix_init(void)
...
-}
+__attribute__((weak)) void matrix_init_kb(void) { matrix_init_user(); }
+
+__attribute__((weak)) void matrix_scan_kb(void) { matrix_scan_user(); }
+
+__attribute__((weak)) void matrix_init_user(void) {}
+
+__attribute__((weak)) void matrix_scan_user(void) {}
+
+__attribute__((weak)) void matrix_slave_scan_user(void) {}
```
Comment (zvecr) [inline keyboards/hhkb/ansi/matrix.c:84]: Only required for split keyboards
```suggestion
```

**Comment by zvecr** [comment]
`hhkb/jp` looks like it also has the same issue, can we apply the same fix there too please.

**Comment by joshuarubin** [comment]
OK, updated in `hhkb/jp` now too

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**